src/huggingface_hub/file_download.py

```diff
@@ -558,7 +558,7 @@ def cached_download(
     token: Union[bool, str, None] = None,
     local_files_only: bool = False,
     legacy_cache_layout: bool = False,
-) -> Optional[str]:  # pragma: no cover
```

Comment:
unrelated to the fix so I can remove it if you strongly feel against it but otherwise I'll keep it (it just fix a type annotation, without any breaking change)